src/potions.ts

```diff
@@ -460,18 +482,64 @@ const rufusPotion = new Potion($item`closed-circuit pay phone`, {
   },
 });
 
-export const wishPotions = Effect.all()
-  .filter((effect) => !effect.attributes.includes("nohookah"))
+export const wishPotions = validWishes.map(
+  (effect) =>
+    new Potion($item`pocket wish`, {
+      effect,
+      canDouble: false,
+      duration: 20,
+      use: (quantity: number) => {
+        const s = getValidWishString(effect);
```

Comment:
you don't need this for genie, afaik; I don't know why I didn't notice this earlier lol

Reply:
if you do need special logic for genie strings, it will be different from the special logic needed for monkey strings, because life is hell

@s-k-z knows a lot about this iirc, and may have an open libram PR that I should merge